Update docs to reflect new framework changes (#44)
dls-controls/tickit: docs/tutorials/creating-an-adapter.rst

Using the Adapter
-----------------

> As explained in `Creating a Device`, an adapter allows us to to control a device

Reviewer: ```suggestion
As explained in `Creating a Device`, an adapter allows us to control a device
```

externally. And in order to use the device we created a `ComponentConfig` for said
device. This is defined in the same file as the device and adapter, and defines any

> default initial configuration values. As well as this, we overwrite the magic method

Reviewer: Jargon...

```suggestion
default initial configuration values. As well as this, we override the magic method
```

`__call__()`, which return a `DeviceSimulation` object. In the `DeviceSimulation`
object, we also define the device's adapter(s):

.. code-block:: python

    @dataclass
    class Shutter(ComponentConfig):

        default_position: float
        initial_position: Optional[float] = None
        host: str = "localhost"
        port: int = 25565

        def __call__(self) -> Component:
            return DeviceSimulation(
                name=self.name,
                device=ShutterDevice(
                    default_position=self.default_position,
                    initial_position=self.initial_position,
                ),
                adapters=[ShutterAdapter(host=self.host, port=self.port)],
            )

> As the adapter(s) are defined in the `ComponentConfig`, they are not required to be
> declared in the YAML for the simulation.

Reviewer: I think this is slightly misleading as it implies you _can_ still do it the old way. I'm not actually sure if you can or can't but either way it's not supported. Maybe:

```suggestion
Using `ComponentConfig` the adapters are instantiated in Python so they are not referenced in the YAML config.
```
